src/main/java/com/twilio/http/bearertoken/BearerTokenTwilioRestClient.java

```diff
@@ -119,9 +128,15 @@ public Response request(BearerTokenRequest request) {
         }
         logRequest(request);
         Response response = httpClient.reliableRequest(request);
+        int statusCode = response.getStatusCode();
+        if(statusCode == 401){
+            this.accessToken = tokenManager.fetchAccessToken();
```

Comment:
Can you test a scenario where fetching of access token is failed.
In my opinion TokenCreator will throw the exception and that will be propagated to orgs api, that is not good, propagating one api exception to another api.

You need to handle this in TokenManager implementation class.